src/storage/query/GetDstBySrcProcessor.cpp

```diff
@@ -28,38 +29,48 @@ void GetDstBySrcProcessor::process(const cpp2::GetDstBySrcRequest& req) {
 }
 
 void GetDstBySrcProcessor::doProcess(const cpp2::GetDstBySrcRequest& req) {
-  if (req.common_ref().has_value() && req.get_common()->profile_detail_ref().value_or(false)) {
-    profileDetailFlag_ = true;
-    profileDetail("GetDstBySrcProcessorTotal", 0);
-    profileDetail("GetDstBySrcProcessorDedup", 0);
-  }
+  try {
+    if (req.common_ref().has_value() && req.get_common()->profile_detail_ref().value_or(false)) {
+      profileDetailFlag_ = true;
+      profileDetail("GetDstBySrcProcessorTotal", 0);
+      profileDetail("GetDstBySrcProcessorDedup", 0);
+    }
 
-  spaceId_ = req.get_space_id();
-  auto retCode = getSpaceVidLen(spaceId_);
-  if (retCode != nebula::cpp2::ErrorCode::SUCCEEDED) {
-    for (auto& p : req.get_parts()) {
-      pushResultCode(retCode, p.first);
+    spaceId_ = req.get_space_id();
+    auto retCode = getSpaceVidLen(spaceId_);
+    if (retCode != nebula::cpp2::ErrorCode::SUCCEEDED) {
+      for (auto& p : req.get_parts()) {
+        pushResultCode(retCode, p.first);
+      }
+      onFinished();
+      return;
     }
-    onFinished();
-    return;
-  }
-  this->planContext_ = std::make_unique<PlanContext>(
-      this->env_, spaceId_, this->spaceVidLen_, this->isIntId_, req.common_ref());
-
-  // check edgetypes exists
-  retCode = checkAndBuildContexts(req);
-  if (retCode != nebula::cpp2::ErrorCode::SUCCEEDED) {
-    for (auto& p : req.get_parts()) {
-      pushResultCode(retCode, p.first);
+    this->planContext_ = std::make_unique<PlanContext>(
+        this->env_, spaceId_, this->spaceVidLen_, this->isIntId_, req.common_ref());
+
+    // check edgetypes exists
+    retCode = checkAndBuildContexts(req);
+    if (retCode != nebula::cpp2::ErrorCode::SUCCEEDED) {
+      for (auto& p : req.get_parts()) {
+        pushResultCode(retCode, p.first);
+      }
+      onFinished();
+      return;
     }
```

Comment:
Could we try-catch it outside the function doProcess in  GraphStorageServiceHandler.cpp?
```
try {
  doProcess()
} catch xxx
```

Reply:
> Could we try-catch it outside the function doProcess in GraphStorageServiceHandler.cpp?
> 
> ```
> try {
>   doProcess()
> } catch xxx
> ```

The call stack is like this, xxxProcessor.process is called from folly
![Screenshot from 2022-12-22 13-13-17](https://user-images.githubusercontent.com/1256658/209310980-7bb53321-f515-4770-ab1f-c65a7915dd95.png)